source/server/listener_manager_impl.cc

```diff
@@ -393,6 +393,17 @@ bool ListenerManagerImpl::addOrUpdateListenerInternal(
   auto existing_active_listener = getListenerByName(active_listeners_, name);
   auto existing_warming_listener = getListenerByName(warming_listeners_, name);
 
+  // The listener should be updated back to its original state and the warming listener should be
```

Comment:
Can you add a `listener_manager_impl_test` unit test for this?

Reply:
added!